src/transformers/models/videomae/modeling_videomae.py

```diff
@@ -859,6 +863,10 @@ def forward(
                     tubelet_size * patch_size * patch_size * num_channels,
                 )
             else:
+                if self.config.num_channels != 3:
+                    raise ValueError(
+                        "Can't unnormalize non-RGB images. Consider setting config.norm_pix_loss to False."
```

Comment:
I feel a bit confused here. We are in the else block, which means we are already self.config.norm_pix_loss == False..
But the message says Consider setting config.norm_pix_loss to False. .... Also this block doesn't have normalization, and the block if self.config.norm_pix_loss: has # step 4: normalize.

@NielsRogge @layjain Could you double check?

Reply:
@ydshieh yes I think this `ValueError` should be raised in case `self.config.norm_pix_loss is True`

Reply:
@ydshieh @NielsRogge  thank you for pointing this out: I'm sorry the error message should have said "Consider setting config.norm_pix_loss to True."

However, I still think that the logic is correct (except for the error message). Here is an explanation:

- If `self.config.num_channels==3`, the original logic is reproduced.
- If `self.config.num_channels != 3` the frames cannot be unnormalized (lines 822-830), therefore, one must use the normalized pixel loss (One must set `config.norm_pix_loss=True`)

> ... Also this block doesn't have normalization, and the block if self.config.norm_pix_loss: has # step 4: normalize.

Regarding this: this block doesn't need normalization because we have never unnormalized (lines 822-824) and hence the frames are already normalized.

Please let me know if you still think there is a problem!

Reply:
Ok, so we should just update the error message False => True

Reply:
Thank you for the explanation @layjain . Sound good to me, would you like to open a PR to correct the error message?

(And by the way, could you show me which part in the else block (lines 866-888) is `unnormalization` ?)